core/src/types.ts

```diff
@@ -159,19 +159,21 @@ export declare interface SurplusAuctionBase {
     network: string;
 }
 
-export declare interface SurplusAuctionActive extends SurplusAuctionBase {
-    bidAmountMKR: BigNumber;
-    receiveAmountDAI: BigNumber;
-    receiverAddress: string;
-    auctionEndDate: Date;
+export declare interface SurplusAuction extends SurplusAuctionBase {
+    state: 'just-started' | 'have-bids' | 'ready-for-collection' | 'requires-restart' | 'collected';
+    fetchedAt: Date;
+    bidAmountMKR?: BigNumber;
+    receiveAmountDAI?: BigNumber;
+    receiverAddress?: string;
+    auctionEndDate?: Date;
     bidEndDate?: Date;
-    earliestEndDate: Date;
-    state: 'just-started' | 'have-bids' | 'ready-for-collection' | 'requires-restart';
-}
-
-export declare interface SurplusAuctionCollected extends SurplusAuctionBase {
-    state: 'collected';
+    earliestEndDate?: Date;
 }
 
-export type SurplusAuction = SurplusAuctionActive | SurplusAuctionCollected;
 export type SurplusAuctionStates = SurplusAuction['state'];
+
+export declare interface SurplusAuctionTransaction extends SurplusAuction {
+    marketUnitPrice?: BigNumber | string;
+    marketUnitPriceToUnitPriceRatio?: BigNumber;
+    unitPrice?: BigNumber;
```

Comment:
not sure what's the difference between this one and the `marketUnitPrice`. Could you clarify? or perhaps @LukSteib could?

Reply:
marketUnitPrice is how much 1 MKR costs on UniSwap
unitPrice is how much 1 MKR costs in the current bidding cycle. 

We use these two values to then calculate the marketUnitPriceToUnitPriceRatio. It helps the user see if the current bid is worth it.

Reply:
ok... so

if i buy 1 mkr for 2 dai on uniswap, and on auction i buy 1 mkr for 1,5 dai, then
```
    marketUnitPrice: 2
    marketUnitPriceToUnitPriceRatio: 1.42
    unitPrice: 1.5
```
right?

Reply:
This is the logic we currently use for collateral auctions:

```ts
const marketUnitPrice = await getMarketPrice(network, collateralSymbol, collateralAmount);
const marketUnitPriceToUnitPriceRatio = approximateUnitPrice
            .minus(marketUnitPrice)
            .dividedBy(marketUnitPrice);
``` 

UnitPrice is fetched from the auction directly, so I assume it should be the same as `totalAmount` / `totalPrice`.

Therefore in your example it should be as follows:

1.5 - 2 = -0.5
-0.5 / 2 = -0.25

Meaning that the current unit price of the auction is -0.25 under the price on Uniswap. I hope this explantation made a bit more sense 🤗

Reply:
ok yes, thanks!